src/components/FilterableProductTable/index.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
+import ProductTable from '../ProductTable';
+import SearchBar from '../SearchBar';
+import CheckBox from '../CheckBox';
+import './styles.css';
+
+export default class FilterableProductTable extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      products: this.props.products,
+      search: '',
+      isChecked: true,
+    };
+  }
+
+  handleInput = (e) => {
+    this.setState({
+      search: e.target.value,
+    });
+  };
+
+  handleCheckBox = (e) => {
```

Comment:
si vas a usar un valor del estado en un setState en class Component usa un callback mejor

`this.setState(state => ({ isChecked: !state.isChecked }))`